lib/components/pagination.vue

```diff
@@ -81,219 +87,286 @@
                @keydown.space.prevent="setPage($event, currentPage + 1)"
-            ><span aria-hidden="true">&raquo;</span></a>
+            ><span aria-hidden="true" v-html="nextText"></span></a>
         </li>
 
     </ul>
 </template>
 
 <script>
-    function isVisible(el) {
-        return el && (el.offsetWidth > 0 || el.offsetHeight > 0);
-    }
+// Determine if an HTML element is visible - Faster than CSS check
+function isVisible(el) {
+    return el && (el.offsetWidth > 0 || el.offsetHeight > 0);
+}
 
-    export default {
-        data() {
-            return {
-                diff: 1,
-                showPrev: false,
-                showNext: false,
-                currentPage: this.value
-            };
+export default {
+    data() {
+        return {
+            offset: 1,
+            showFirstPage: false,
+            showFirstDots: false,
+            showLastPage: false,
+            showLastDots: false,
+            currentPage: this.value
+        };
+    },
+    computed: {
+        numberOfPages() {
+            const result = Math.ceil(this.totalRows / this.perPage);
+            return (result < 1) ? 1 : result;
         },
-        computed: {
-            numberOfPages() {
-                const result = Math.ceil(this.totalRows / this.perPage);
-                return (result < 1) ? 1 : result;
-            },
-            btnSize() {
-                return this.size ? `pagination-${this.size}` : '';
-            },
-            pageLinks() {
-                if (this.currentPage > this.numberOfPages) {
-                    this.currentPage = 1;
-                }
+        btnSize() {
+            return this.size ? `pagination-${this.size}` : '';
+        },
+        pageLinks() {
+            // Sanity checks
+            if (this.currentPage > this.numberOfPages) {
+                this.currentPage = this.numberOfPages;
+            } else if (this.currentPage < 1) {
+                this.currentPage = 1;
+            }
 
-                // Defaults
-                this.diff = 1;
-                this.showPrev = false;
-                this.showNext = false;
+            // Defaults
+            this.offset = 1;
+            this.showFirstPage = false;
+            this.showFirstDots = false;
+            this.showLastPage = false;
+            this.showLastDots = false;
 
-                // If less pages than limit just show this pages
-                if (this.numberOfPages <= this.limit) {
-                    return this.numberOfPages;
-                }
+            // Special case
+            if (this.numberOfPages <= this.limit) {
+                return this.numberOfPages;
+            }
 
-                // If at the beginning of the list
-                if (this.currentPage <= this.limit - 2) {
-                    this.diff = 1;
-                    this.showNext = true;
-                    return this.limit - 2;
-                }
+            // Special case
```

Comment:
Still trying to figure this one out. Also curious if it should take the number of pages into account.